cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringAgentService.java

```diff
@@ -158,14 +214,11 @@ protected long runTask() {
     }
 
     // Update last message ids in the store for all peers
-    TransactionRunners.run(transactionRunner, context -> {
-      AppMetadataStore appMetadataStore = AppMetadataStore.create(context);
-      for (Map.Entry<String, String> entry : lastMessageIds.entrySet()) {
-        appMetadataStore.persistSubscriberState(entry.getKey(), SUBSCRIBER, entry.getValue());
-      }
-    });
-
-    return connectionInterval;
+    AppMetadataStore appMetadataStore = getAppMetadataStore(context);
+    for (Map.Entry<String, String> entry : peerToLastControlMessageIds.entrySet()) {
+      appMetadataStore.persistSubscriberState(entry.getKey(), SUBSCRIBER, entry.getValue());
```

Comment:
this is normally used when we read from a topic, using the subscriber as the identifier for the reader. If we store the peer name as the "topic", it can override something important if the peer name happens to be the same as some internal topic name. To prevent these clashes, we need to use some constant prefix like "peer.program.state." so that the topic becomes "peer.program.state.[peer name]"

Reply:
fixed